hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyPurgeRequest.java

```diff
@@ -51,17 +54,28 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
     PurgeKeysRequest purgeKeysRequest = getOmRequest().getPurgeKeysRequest();
     List<DeletedKeys> bucketDeletedKeysList = purgeKeysRequest
         .getDeletedKeysList();
-    List<String> keysToBePurgedList = new ArrayList<>();
+    List<Pair<OmBucketInfo, List<String>>> keysToBePurgedList
+        = new ArrayList<>();
 
     OMResponse.Builder omResponse = OmResponseUtil.getOMResponseBuilder(
         getOmRequest());
     OMClientResponse omClientResponse = null;
 
 
+    OMMetadataManager omMetadataManager = ozoneManager.getMetadataManager();
     for (DeletedKeys bucketWithDeleteKeys : bucketDeletedKeysList) {
+      String volumeName = bucketWithDeleteKeys.getVolumeName();
+      String bucketName = bucketWithDeleteKeys.getBucketName();
+      OmBucketInfo omBucketInfo = getBucketInfo(omMetadataManager, volumeName,
```

Comment:
Since we are updating bucketTable metadata here , bucket lock needs to be acquired, else someone can delete the bucket.